clang/lib/CodeGen/Targets/NVPTX.cpp

```diff
@@ -218,6 +222,98 @@ Address NVPTXABIInfo::EmitVAArg(CodeGenFunction &CGF, Address VAListAddr,
   llvm_unreachable("NVPTX does not support varargs");
 }
 
+// Get current CudaArch and ignore any unknown values
+// Copied from CGOpenMPRuntimeGPU
+static CudaArch getCudaArch(CodeGenModule &CGM) {
+  if (!CGM.getTarget().hasFeature("ptx"))
+    return CudaArch::UNKNOWN;
+  for (const auto &Feature : CGM.getTarget().getTargetOpts().FeatureMap) {
+    if (Feature.getValue()) {
+      CudaArch Arch = StringToCudaArch(Feature.getKey());
+      if (Arch != CudaArch::UNKNOWN)
+        return Arch;
+    }
+  }
+  return CudaArch::UNKNOWN;
+}
+
+static bool supportsGridConstant(CudaArch Arch) {
+  switch (Arch) {
+  case CudaArch::SM_70:
+  case CudaArch::SM_72:
+  case CudaArch::SM_75:
+  case CudaArch::SM_80:
+  case CudaArch::SM_86:
+  case CudaArch::SM_87:
+  case CudaArch::SM_89:
+  case CudaArch::SM_90:
+  case CudaArch::SM_90a:
+    return true;
+  case CudaArch::UNKNOWN:
+  case CudaArch::UNUSED:
+  case CudaArch::SM_20:
+  case CudaArch::SM_21:
+  case CudaArch::SM_30:
+  case CudaArch::SM_32_:
+  case CudaArch::SM_35:
+  case CudaArch::SM_37:
+  case CudaArch::SM_50:
+  case CudaArch::SM_52:
+  case CudaArch::SM_53:
+  case CudaArch::SM_60:
+  case CudaArch::SM_61:
+  case CudaArch::SM_62:
+    return false;
+  case CudaArch::GFX600:
+  case CudaArch::GFX601:
+  case CudaArch::GFX602:
+  case CudaArch::GFX700:
+  case CudaArch::GFX701:
+  case CudaArch::GFX702:
+  case CudaArch::GFX703:
+  case CudaArch::GFX704:
+  case CudaArch::GFX705:
+  case CudaArch::GFX801:
+  case CudaArch::GFX802:
+  case CudaArch::GFX803:
+  case CudaArch::GFX805:
+  case CudaArch::GFX810:
+  case CudaArch::GFX900:
+  case CudaArch::GFX902:
+  case CudaArch::GFX904:
+  case CudaArch::GFX906:
+  case CudaArch::GFX908:
+  case CudaArch::GFX909:
+  case CudaArch::GFX90a:
+  case CudaArch::GFX90c:
+  case CudaArch::GFX940:
+  case CudaArch::GFX941:
+  case CudaArch::GFX942:
+  case CudaArch::GFX1010:
+  case CudaArch::GFX1011:
+  case CudaArch::GFX1012:
+  case CudaArch::GFX1013:
+  case CudaArch::GFX1030:
+  case CudaArch::GFX1031:
+  case CudaArch::GFX1032:
+  case CudaArch::GFX1033:
+  case CudaArch::GFX1034:
+  case CudaArch::GFX1035:
+  case CudaArch::GFX1036:
+  case CudaArch::GFX1100:
+  case CudaArch::GFX1101:
+  case CudaArch::GFX1102:
+  case CudaArch::GFX1103:
+  case CudaArch::GFX1150:
+  case CudaArch::GFX1151:
+  case CudaArch::GFX1200:
+  case CudaArch::GFX1201:
+  case CudaArch::Generic:
+  case CudaArch::LAST:
+    llvm_unreachable("unhandled CudaArch");
+  }
```

Comment:
```suggestion
  if (IsNVIDIAGpuArch(Arch) && Arch >= CudaArch::SM_70)
    return true;
  return false;
```

or 

```suggestion
  assert(IsNVIDIAGpuArch(Arch));
  if (Arch >= CudaArch::SM_70)
    return true;
  return false;
```

(`assert` is probably better than the `llvm_unreachable `)

Reply:
Whoops, yeah. I was so focused on the existing patterns of explicitly listing out every single enum value (I guess to ensure the code is correctly inspected if any enum members are added or removed) that I missed that this particular case doesn't need that care. That is, assuming that grid_constant will be available on all architectures >= SM_70. That's probably safe enough a guess for now.